integration_tests/examples/demo/utils/inputs/rtp.rs

```diff
@@ -0,0 +1,235 @@
+use anyhow::{anyhow, Result};
+use inquire::Select;
+use integration_tests::{
+    ffmpeg::start_ffmpeg_send,
+    gstreamer::{start_gst_send_tcp, start_gst_send_udp},
+};
+use serde_json::json;
+use strum::{Display, EnumIter, IntoEnumIterator};
+use tracing::error;
+
+use crate::utils::{
+    get_free_port,
+    inputs::{input_name, AudioDecoder, InputHandler, VideoDecoder},
+    TransportProtocol, IP,
+};
+
+#[derive(Debug, Display, EnumIter, Clone)]
+pub enum RtpRegisterOptions {
+    #[strum(to_string = "Add video stream")]
+    AddVideoStream,
+
+    #[strum(to_string = "Add audio stream")]
+    AddAudioStream,
+
+    #[strum(to_string = "Remove video stream")]
+    RemoveVideoStream,
+
+    #[strum(to_string = "Remove audio stream")]
+    RemoveAudioStream,
+
+    #[strum(to_string = "Set transport protocol")]
+    SetTransportProtocol,
+
+    #[strum(to_string = "Done")]
+    Done,
+}
+
+#[derive(Debug)]
+pub struct RtpInput {
+    name: String,
+    port: u16,
+    video: Option<RtpInputVideoOptions>,
+    audio: Option<RtpInputAudioOptions>,
+    transport_protocol: TransportProtocol,
+}
+
+impl RtpInput {
+    pub fn setup() -> Result<Self> {
+        let mut rtp_input = Self {
+            name: input_name(),
+
+            // TODO: (@jbrs) Make it possible for user
+            // to set their own port
+            port: get_free_port(),
+            video: None,
+            audio: None,
+            transport_protocol: TransportProtocol::Udp,
+        };
+
+        let options = RtpRegisterOptions::iter().collect::<Vec<_>>();
+        loop {
+            let action = Select::new("What to do?", options.clone()).prompt()?;
+
+            match action {
+                RtpRegisterOptions::AddVideoStream => rtp_input.setup_video()?,
```

Comment:
why are those things action that user can select? When registering input/output just ask everything in order with option to skip if optional.